packages/java/parser-jvm-plugin-nonnull-kotlin/src/main/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/KotlinNullabilityPlugin.kt

```diff
@@ -0,0 +1,258 @@
+        return nodeDependencies.processChildNodes { it.sorted(childNodesComparator) }
+    }
+
+    private fun getKotlinPropertyNames(clazz: Class<*>): List<String> {
+        val fields = clazz.declaredFields.map { it.name.substringBefore("\$delegate") }
+
+        val getters = clazz.declaredMethods
+            .filter { it.name.startsWith("get") && it.parameterCount == 0 }
+            .map { it.name.removePrefix("get").replaceFirstChar(Char::lowercase) }
+
+        val setters = clazz.declaredMethods
+            .filter { it.name.startsWith("set") && it.parameterCount == 1 }
+            .map { it.name.removePrefix("set").replaceFirstChar(Char::lowercase) }
+
+        return (fields + getters + setters).distinct()
+    }
+
+    override fun enter(nodePath: NodePath<*>?) {
+        // No action needed on enter
+    }
+
+    override fun resolve(node: Node<*, *>, parentPath: NodePath<*>): Node<*, *> {
+        // If node is already a Kotlin node, nothing to do.
+        if (node is KNode) return node
+
+        // Check that the parent class is Kotlin. If not, return unchanged.
+        val parentClass: Class<*>? = when (node) {
+            is EndpointNode, is EntityNode -> (node.source as? ClassInfoModel)?.get() as? Class<*>
+            is MethodNode, is PropertyNode -> (parentPath.node.source as? ClassInfoModel)?.get() as? Class<*>
+            else -> null
+        }
+        if (parentClass != null && !isKotlinClass(parentClass)) return node
+        if (parentPath != parentPath.rootPath && !isKotlinClass(findClosestClass(parentPath))) return node
+
+        return when (node) {
+            is EndpointNode -> KEndpointNode(node.source, node.target, (node.source.get() as Class<*>).kotlin)
+            is EndpointExposedNode -> KEndpointExposedNode(node.source, (node.source.get() as Class<*>).kotlin)
+            is MethodNode -> createKMethodNode(node, parentPath)
+            is MethodParameterNode -> KMethodParameterNode(node.source, node.target,
+                    (parentPath.node as KMethodNode).kFunction.parameters
+                        .first { it.kind == KParameter.Kind.VALUE && it.name == node.source.name })
+            is TypedNode -> resolveTypedNode(node, parentPath)
+            is EntityNode ->
+                KEntityNode(node.source, node.target as ObjectSchema, (node.source.get() as Class<*>).kotlin)
+            is PropertyNode -> {
+                val kProperty = (parentPath.node as? KEntityNode)?.kClass?.memberProperties
+                    ?.firstOrNull { it.name == node.source.name }
+                if (kProperty != null) {
+                    KPropertyNode(node.source, node.target, kProperty)
+                } else {
+                    // If the property is not found, it possibly can be a property defined using getter/setter methods
+                    // then leave it unchanged (will be handled in exit)
+                    node
+                }
```

Comment:
```suggestion
                kProperty?.let { KPropertyNode(node.source, node.target, it) } ?: node
```
(preserving the comment of course)